routing/router.go

```diff
@@ -1199,9 +1207,9 @@ func (r *ChannelRouter) sendToRoute(htlcHash lntypes.Hash, rt *route.Route,
 	// An error returned from collecting the result, we'll mark the payment
 	// as failed if we don't skip temp error.
 	if !skipTempErr {
-		err := r.cfg.Control.FailPayment(paymentIdentifier, reason)
+		err := failPayment(paymentIdentifier, reason)
```

Comment:
I think we should update the test to check the new behavior that we won't fail twice.

Reply:
you mean the expectations ? Yeah updated them to `Once`